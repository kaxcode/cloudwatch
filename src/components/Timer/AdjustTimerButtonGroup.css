.AdjustTimerButtonGroup__PlusBtn {
  margin-right: 20px;
  margin-left: 20px;
}

.AdjustTimerButtonGroup {
  grid-area: adjustBtns;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}
