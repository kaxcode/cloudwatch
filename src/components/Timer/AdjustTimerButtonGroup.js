import React from 'react';
import { Button } from 'react-materialize';
import { func } from 'prop-types';

const AdjustTimerButtonGroup = props => {
  return (
    <tr>
      <td>
        <div className="hours-set">
          <Button
            id="hours-increase"
            className="plus-btn blue"
            onClick={props.increaseHours}
          >
            +
          </Button>
          <Button
            id="hours-decrease"
            className="minus-btn blue"
            onClick={props.decreaseHours}
          >
            -
          </Button>
        </div>
      </td>
      <td>
        <div className="minutes-set">
          <Button
            id="minutes-increase"
            className="plus-btn blue"
            onClick={props.increaseMinutes}
          >
            +
          </Button>
          <Button
            id="minutes-decrease"
            className="minus-btn blue"
            onClick={props.decreaseMinutes}
          >
            -
          </Button>
        </div>
      </td>
      <td>
        <div className="seconds-set">
          <Button
            waves="light"
            id="seconds-increase"
            className="plus-btn blue"
            onClick={props.increaseSeconds}
          >
            +
          </Button>
          <Button
            id="seconds-decrease"
            className="minus-btn blue"
            onClick={props.decreaseSeconds}
          >
            -
          </Button>
        </div>
      </td>
    </tr>
  );
};

AdjustTimerButtonGroup.propTypes = {
  increaseHours: func.isRequired,
  decreaseHours: func.isRequired,
  increaseMinutes: func.isRequired,
  decreaseMinutes: func.isRequired,
  increaseSeconds: func.isRequired,
  decreaseSeconds: func.isRequired
};

export default AdjustTimerButtonGroup;
