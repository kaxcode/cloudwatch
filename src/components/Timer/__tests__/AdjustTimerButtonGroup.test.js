import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import AdjustTimerButtonGroup from '../AdjustTimerButtonGroup';

Enzyme.configure({ adapter: new Adapter() });

describe('Adjudst Timer Button Group', () => {
  let wrapper;

  const increaseHoursMock = jest.fn();
  const decreaseHoursMock = jest.fn();
  const increaseMinutesMock = jest.fn();
  const decreaseMinutesMock = jest.fn();
  const increaseSecondsMock = jest.fn();
  const decreaseSecondsMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <AdjustTimerButtonGroup
        increaseHours={increaseHoursMock}
        decreaseHours={decreaseHoursMock}
        increaseMinutes={increaseMinutesMock}
        decreaseMinutes={decreaseMinutesMock}
        increaseSeconds={increaseSecondsMock}
        decreaseSeconds={decreaseSecondsMock}
      />
    );
  });
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('does not render if in presenter window', () => {
    global.window.name = 'presenter';
    wrapper = shallow(
      <AdjustTimerButtonGroup
        increaseHours={increaseHoursMock}
        decreaseHours={decreaseHoursMock}
        increaseMinutes={increaseMinutesMock}
        decreaseMinutes={decreaseMinutesMock}
        increaseSeconds={increaseSecondsMock}
        decreaseSeconds={decreaseSecondsMock}
      />
    );
    wrapper.update();
    expect(wrapper.type()).toEqual(null);
  });
});
