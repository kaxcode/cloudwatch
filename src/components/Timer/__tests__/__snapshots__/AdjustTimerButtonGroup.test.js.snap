// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adjudst Timer Button Group renders correctly 1`] = `
<div
  className="AdjustTimerButtonGroup"
>
  <div
    className="Timer__HoursAdjustBtn__Container"
  >
    <i
      className="fas fa-angle-up AdjustTimerButtonGroup__PlusBtn"
      id="hours-increase"
      onClick={[Function]}
    />
    <i
      className="fas fa-angle-down AdjustTimerButtonGroup__MinusBtn"
      id="hours-decrease"
      onClick={[Function]}
    />
  </div>
  <div
    className="Timer__MinutesAdjustBtn__Container"
  >
    <i
      className="fas fa-angle-up AdjustTimerButtonGroup__PlusBtn"
      id="minutes-increase"
      onClick={[Function]}
    />
    <i
      className="fas fa-angle-down AdjustTimerButtonGroup__MinusBtn"
      id="minutes-decrease"
      onClick={[Function]}
    />
  </div>
  <div
    className="Timer__SecondsAdjustBtn__Container"
  >
    <i
      className="fas fa-angle-up AdjustTimerButtonGroup__PlusBtn"
      id="seconds-increase"
      onClick={[Function]}
    />
    <i
      className="fas fa-angle-down AdjustTimerButtonGroup__MinusBtn"
      id="seconds-decrease"
      onClick={[Function]}
    />
  </div>
</div>
`;
